validator/sawtooth_validator/config/validator.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import collections
+import logging
+import os
+
+import toml
+
+from sawtooth_validator.exceptions import LocalConfigurationError
+from sawtooth_validator.config.path import get_default_path_config
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def load_toml_validator_config(filename):
```

Comment:
Would it be clearer if the top level methods were instead class methods of  ValidatorConfig?

```
load_toml_validator_config -> ValidatorConfig.from_file()
merge_path_config -> ValidatorConfig.merge()
+ ValidatorConfig.from_opts()
```

If the default behaviour of from_file() is then to load from default_path_config.config_dir, then you have:
```
def load_validator_config(opts, config_dir=None):
  opt_conf = ValidatorConfig.from_opts(opts)
  file_conf = ValidatorConfig.from_file(config_dir)
  return ValidatorConfig.merge(opt_conf, file_conf)
```